clang/test/SemaSYCL/intel-max-global-work-dim-device.cpp

```diff
@@ -101,17 +103,12 @@ struct TRIFuncObjGood8 {
   operator()() const {}
 };
 
-// FIXME: We do not have support yet for checking
-// max_work_group_size and max_global_work_dim
-// attributes when merging, so the test compiles without
-// any diagnostic when it shouldn't.
-struct TRIFuncObjBad {
-  [[intel::max_work_group_size(4, 4, 4)]] void
+struct TRIFuncObjGood9 {
+  [[intel::max_work_group_size(4, 4, 4)]] void // OK
   operator()() const;
 };
 
-[[intel::max_global_work_dim(0)]]
-void TRIFuncObjBad::operator()() const {}
+[[intel::max_global_work_dim(1)]] void TRIFuncObjGood9::operator()() const {}
 
 // FIXME: We do not have support yet for checking
```

Comment:
Is this one still needed?

Reply:
This is still needed because we are checking for reqd_work_group_size and max_global_work_dim attributes here when merging. Previous comment was wring. I have updated now. Thanks @premanandrao for pointing this.